pyannote/audio/applications/multilabel.py

```diff
@@ -0,0 +1,584 @@
+
+from pyannote.audio.features import Precomputed
+
+from pyannote.core.utils.helper import get_class_by_name
+from pyannote.core import Timeline, SlidingWindowFeature
+from pyannote.audio.labeling.tasks import Multilabel as MultilabelTask
+
+
+def validate_helper_func(current_file, pipeline=None, precision=None, recall=None, label=None, metric=None):
+    reference = current_file[label+"_ref"]
+    # pipeline has been initialized with label, so that it can know which class needs to be assessed
+    hypothesis = pipeline(current_file)
+    uem = get_annotated(current_file)
+
+    if precision is not None:
+        p = precision(reference, hypothesis, uem=uem)
+        r = recall(reference, hypothesis, uem=uem)
+        return p, r
+    else:
+        return metric(reference, hypothesis, uem=uem)
+
+
+class Multilabel(BaseLabeling):
+
+    def __init__(self, protocol_name, experiment_dir, db_yml=None, training=False, use_der=False):
+        super(BaseLabeling, self).__init__(
+            experiment_dir, db_yml=db_yml, training=training)
+
+        self.use_der = use_der
+
+        # task
+        Task = get_class_by_name(
+            self.config_['task']['name'],
+            default_module_name='pyannote.audio.labeling.tasks')
+
+        self.task_ = Task(protocol_name=protocol_name,
+                          preprocessors=self.preprocessors_,
+                          **self.config_['task'].get('params', {}))
+
+        # architecture
+        Architecture = get_class_by_name(
+            self.config_['architecture']['name'],
+            default_module_name='pyannote.audio.labeling.models')
+
+        params = self.config_['architecture'].get('params', {})
+        self.get_model_ = partial(Architecture, **params)
+
+        if hasattr(Architecture, 'get_frame_info'):
+            self.frame_info_ = Architecture.get_frame_info(**params)
+        else:
+            self.frame_info_ = None
+
+        if hasattr(Architecture, 'frame_crop'):
+            self.frame_crop_ = Architecture.frame_crop
+        else:
+            self.frame_crop_ = None
+
+    @classmethod
+    def from_train_dir(cls, protocol_name, train_dir, db_yml=None, training=False, use_der=False):
+        experiment_dir = dirname(dirname(train_dir))
+        app = cls(protocol_name, experiment_dir, db_yml=db_yml, training=training, use_der=use_der)
+        app.train_dir_ = train_dir
+
+        return app
+
+    @classmethod
+    def from_model_pt(cls, protocol_name, model_pt, db_yml=None, training=False, use_der=False):
+        train_dir = dirname(dirname(model_pt))
+        app = cls.from_train_dir(protocol_name, train_dir, db_yml=db_yml, training=training, use_der=use_der)
+        app.model_pt_ = model_pt
+        epoch = int(basename(app.model_pt_)[:-3])
+        app.model_ = app.load_model(epoch, train_dir=train_dir)
+        return app
+
+    def validate_init(self, protocol_name, subset='development'):
```

Comment:
Can we call `validation_data = super().validate_init(protocol_name, subset=subset)` at the beginning?
This way we can inherit from `BaseLabeling.validate_init` and avoid code duplication.